common/lib/plugins/efm2/host_monitoring2_connection_plugin.ts

```diff
@@ -0,0 +1,175 @@
+    return this.connectInternal(hostInfo, connectFunc);
+  }
+
+  forceConnect(
+    hostInfo: HostInfo,
+    props: Map<string, any>,
+    isInitialConnection: boolean,
+    forceConnectFunc: () => Promise<ClientWrapper>
+  ): Promise<ClientWrapper> {
+    return this.connectInternal(hostInfo, forceConnectFunc);
+  }
+
+  private async connectInternal(hostInfo: HostInfo, connectFunc: () => Promise<ClientWrapper>): Promise<ClientWrapper> {
+    const targetClient = await connectFunc();
+    if (targetClient != null) {
+      const type: RdsUrlType = this.rdsUtils.identifyRdsType(hostInfo.host);
+      if (type.isRdsCluster) {
+        hostInfo.resetAliases();
+        await this.pluginService.fillAliases(targetClient, hostInfo);
+      }
+    }
+    return targetClient;
+  }
+
+  async execute<T>(methodName: string, methodFunc: () => Promise<T>, methodArgs: any): Promise<T> {
+    const isEnabled: boolean = WrapperProperties.FAILURE_DETECTION_ENABLED.get(this.properties) as boolean;
+
+    if (!isEnabled || !SubscribedMethodHelper.NETWORK_BOUND_METHODS.includes(methodName)) {
+      return methodFunc();
+    }
+
+    const failureDetectionTimeMillis: number = WrapperProperties.FAILURE_DETECTION_TIME_MS.get(this.properties) as number;
+    const failureDetectionIntervalMillis: number = WrapperProperties.FAILURE_DETECTION_INTERVAL_MS.get(this.properties) as number;
+    const failureDetectionCount: number = WrapperProperties.FAILURE_DETECTION_COUNT.get(this.properties) as number;
+
+    let result: T;
+    let monitorContext: MonitorConnectionContext | null = null;
+
+    try {
+      logger.debug(Messages.get("HostMonitoringConnectionPlugin.activatedMonitoring", methodName));
+      const monitoringHostInfo: HostInfo = await this.getMonitoringHostInfo();
+
+      monitorContext = await this.monitorService.startMonitoring(
+        this.pluginService.getCurrentClient().targetClient,
+        monitoringHostInfo,
+        this.properties,
+        failureDetectionTimeMillis,
+        failureDetectionIntervalMillis,
+        failureDetectionCount
+      );
+
+      result = await methodFunc();
+    } finally {
+      if (monitorContext != null) {
+        await this.monitorService.stopMonitoring(monitorContext, this.pluginService.getCurrentClient().targetClient);
+
+        logger.debug(Messages.get("HostMonitoringConnectionPlugin.monitoringDeactivated", methodName));
+      }
+    }
+
+    return result;
+  }
+
+  private throwUnableToIdentifyConnection(host: HostInfo | null): never {
+    const provider: HostListProvider | null = this.pluginService.getHostListProvider();
+    throw new AwsWrapperError(
+      Messages.get(
+        "HostMonitoringConnectionPlugin.unableToIdentifyConnection",
+        host != null ? host.host : "unknown host",
+        provider != null ? provider.getHostProviderType() : "unknown provider"
+      )
+    );
+  }
+
+  async getMonitoringHostInfo(): Promise<HostInfo> {
+    if (this.monitoringHostInfo == null) {
```

Comment:
Can we flip the condition and return early if `this.monitoringHostInfo !== null`?
i.e.
```ts
    if (this.monitoringHostInfo !== null) {
      return this.monitoringHostInfo;
    }

    this.monitoringHostInfo = this.pluginService.getCurrentHostInfo();
    if (this.monitoringHostInfo === null) {
      this.throwUnableToIdentifyConnection(null);
    }
...
```